feat(proto): automatically include third party proto

ilgooz opened the pull request:
* removed third party proto paths from the scaffold (third_party/proto and proto/cosmos).
* removed third party proto now automatically included during code generation by using Go's module system. (not by a dark magic 🧙).
* for now scaffolded apps only needs third party protos from Cosmos SDK itself, keep in mind that SDK has a `go.mod` thereof it was possible to use Go's module system to get its proto files (which are `proto/` and `third_party/proto/`).

BE AWARE that:

* if scaffolded apps at some point require some other third party protos from different repos that also have `go.mod`s, we can implement another feature that reads Git urls of these repos from the `proto` section of config.yml` and easily access their protos.

        FUTURE NOTE, following posibilities will occur:
                * third party added to `config.yml` might be imported by the app's `go.mod`
                * third party added to `config.yml` might not be imported by the app's `go.mod`. this also requires version of the third party to be defined in the `config.yml`.
                * relative paths of proto dirs needed under these third party proto repos must be defined in the `config.yml` as well.
                * keep in mind conflicting proto files with the same proto and Go package name.

* proto repos that does not have a `go.mod` (means not a Go project) still can be manually downloaded and moved under `proto_vendor` or `third_party/proto` (supported for backwards compability) paths.
as it is easily posible to use proto files from Go projects, supporting non-Go ones requires some additional effort of developing a minimal package manager for proto since it is not possible to relay this work to Go's module system anymore.

fadeev commented:
I really like not having third party stuff in the repo. This does look a bit like Starport-specific magic, I'm still trying to wrap my head around it 🙂

Gitpod's having problems with this approach, though

```
gitpod /workspace/starport/docs $ starport app github.com/hello/world
google/protobuf/descriptor.proto: File not found.
google/api/annotations.proto:20:1: Import "google/protobuf/descriptor.proto" was not found or had errors.
google/api/annotations.proto:28:8: "google.protobuf.MethodOptions" is not defined.
world/query.proto:4:1: Import "google/api/annotations.proto" was not found or had errors.
: exit status 1
```

ilgooz commented:
Yeah, this PR deletes `google/protobuf/descriptor.proto` from the scaffold. I cannot remember but I guess it was manually placed before into the `third_party` dir or it was there before in the SDK's repo and deleted later on.

Even though it is now deleted, it and actually the whole `google/*` should already be made available under users' machine during the installation of `protoc`. For ex. while installating `protoc` via `brew` or `apt` it should come by default. On manual installations, it is [instructed ](https://grpc.io/docs/protoc-installation/#install-pre-compiled-binaries-any-os)that they should be placed under `/usr/.local/include`.

Turns out, in our manual setup of `protoc` on Gitpod, we had a miss/outdated installation. Fixed by 5f41257.

ilgooz commented:
go-bindata issue fixed by https://github.com/tendermint/starport/pull/686. ⚡

ilgooz commented:
Gitpod is back. 🍰

fadeev commented:
I must admit that this is pretty sweet 🍩

fadeev reviewed (approved):
Awesome 🔥
